packages/react-components/src/components/utils/useLongPress.ts

```diff
@@ -29,38 +29,48 @@ export default function useLongPress(
     setIsLongPress(false);
     timerRef.current = setTimeout(() => {
       setIsLongPress(true);
+      onLongPress();
     }, 500);
   }
 
   function handleOnClick(): void {
-    // when it's mobile use ontouchstart and ontouchend to fire onclick and onlongpress event
     if (isMobile) {
       return;
     }
-    onClick();
-    if (isLongPress) {
-      onLongPress();
-      return;
+    if (!isLongPress) {
+      onClick();
     }
   }
 
   function handleOnKeyDown(): void {
+    if (isMobile) {
```

Comment:
should `handleOnKeyDown` still work if mobile has a bluetooth keyboard plugged in ?

Reply:
This logic seems to be there to make isMobile only trigger for touch events is that right?
Instead would we pass the start mechanism into `startPressTimer` e.g.
```
startPressTimer('key' | 'mouseClick' | 'touch');
```
then the startPressTimer can handle accordingly?
```
...
setTimeout(() => {
  onLongPress();
  if (pressMechanism === 'touch') {
    onLongTouch();
  } 
}
```
?

Reply:
Given that we only have onClick and onLongPress as functionalities in this hook.
I think changing isMobile to touchEventsOnly makes more sense.